content/fr/articles/css/stop-css/index.md

```diff
@@ -0,0 +1,233 @@
+
+Si ce constat peut aussi permettre à ceux et celles qui utilisent des CSS atomiques d'arrêter immédiatement:
+
+```css
+.relative { position: relative; }
+.absolute { position: absolute; }
+.static { position: static; }
+```
+
+Dans le cas précédent, si vous ajoutez une classe `relative` à un élement ayant déjà la classe `static`, ça ne changera rien du tout, parce que `.static` est défini après `.relative`. Génial, non ?
+
+### Le futur de CSS
+
+Les variables sont une feature qui a été très demandée à CSS. Mais cette feature va débarquer avec son lot de souci :
+
+```css
+:root {
+  --mainColor: blue;
+}
+
+body {
+  color: var(--mainColor);
+}
+```
+
+C'est super, mais si quelqu'un vient ajouter :
+
+```css
+:root {
+  --mainColor: red;
+}
+```
+
+quelque part, il y a deux cas:
+
+- soit c'est chargé avant, et ça ne fait rien
+- soit c'est chargé après, et ça override toutes les propriétés utilisant `--mainColor`. On est bien contents.
+
+## Vous ne pouvez pas vous permettre de laisser tomber CSS ?
+
+Dans ce cas, forcez vous à utiliser la [méthodologie BEM](/fr/articles/css/bem/). Ça ne réglera pas tout, mais au moins cela vous permettra d'éviter un maximum de conneries en vous forçant à découper en composants isolés, et à mieux maîtriser la priorisation, puisque le scope auquel vous devrez y veiller sera considérablement réduit:
+
+```css
+/* Header.css */
+.Header {}
+
+  .Header-nav {}
+
+/* Nav.css */
+.Nav {}
+
+  .Nav-item {}
+
+    .Nav-item--active {}
+```
+
+C'est quand même plus pratique à comprendre, non ?
+
+Encore mieux, si vous avez du tooling à la webpack, vous pouvez utiliser les [CSS modules](/fr/articles/css/modules/), qui limiteront de même le scope d'application de vos feuilles de style.
+
+## Vous pouvez vous permettre de laisser tomber CSS ?
+
+JavaScript vous permet déjà de bénéficier d'un système de modules, de variables, de conditions, de fonctions réutilisables, et tout ça sans hack. En plus de ça, vous maitrisez la priorisation, parce c'est que c'est vous qui décidez ce qui s'applique:
+
+```javascript
+class MyComponent extends React.Component {
+  render() {
+    const { active, disabled } = this.props
+    return (
+      <div
+        style={{
+          ...styles.myComponent,
+          ...active ? styles.active : null,
+          ...disabled && !active ? styles.disabled : null,
+        }}
+      >
+        tadaa
+      </div>
+    )
+  }
+}
+
+const styles = {
+  myComponent: {
+    fontSize: 18,
+  },
+  active: {
+    textDecoration: "underline",
+  },
+  disabled: {
+    opacity: 0.5,
+  },
+}
+```
+
+Par ailleurs, avec certaines bibliothèques permettant naturellement l'usage d'inline-styles (comme React), cela vous permet d'avoir le style et le markup dans le même fichier, sans avoir besoin de naviguer entre les onglets de votre éditeur.
```

Comment:
`cela vous permet de garder à proximité les styles lié à un composant, que ce soit dans le même fichier, ou un fichier de style juste à côté de la définition du composant`? Something like that.